packages/dashboard-core-plugins/src/ChartPanelPlugin.tsx

```diff
@@ -69,43 +65,41 @@ async function createChartModel(
 
 export const ChartPanelPlugin = forwardRef(
   (props: WidgetPanelProps<DhType.plot.Figure>, ref: React.Ref<ChartPanel>) => {
-    const deferredApi = useContext(DeferredApiContext);
-
     const panelState = isChartPanelDehydratedProps(props)
       ? (props as unknown as ChartPanelProps).panelState
       : undefined;
 
     const { fetch: panelFetch, metadata, localDashboardId } = props;
-
-    const hydratedProps = useMemo(
-      () => ({
-        metadata: metadata as ChartPanelMetadata,
-        localDashboardId,
-        makeModel: async () => {
-          if (metadata == null) {
-            throw new Error('Metadata is required for chart panel');
-          }
-
-          const dh =
-            typeof deferredApi === 'function'
-              ? await deferredApi(metadata)
-              : deferredApi;
-
-          assertNotNull(dh, `Cannot find API for metadata: ${metadata}`);
-
-          return createChartModel(
-            dh,
-            metadata as ChartPanelMetadata,
-            panelFetch,
-            panelState
-          );
-        },
-      }),
-      [metadata, localDashboardId, deferredApi, panelFetch, panelState]
+    assertNotNull(metadata);
+    const [dh, error] = useDeferredApi(metadata);
+
+    const makeModel = useCallback(async () => {
+      if (error != null) {
+        throw error;
+      }
```

Comment:
Does this error propagate correctly? Just curious as I've had some issues recently w/ throwing from async or effects. Since it executes outside the react render cycle, it doesn't get caught by the error boundary

Reply:
Yea it gets caught by ChartPanel, when it calls `makeModel`